core/nuxeo-operation.js

```diff
@@ -308,20 +309,22 @@ import './nuxeo-connection.js';
           return this.response;
         })
         .catch((error) => {
-          if (error.response.status === 401) {
-            this.dispatchEvent(
-              new CustomEvent('unauthorized-request', {
-                bubbles: true,
-                composed: true,
-                detail: error,
-              }),
-            );
-          }
+          const event = error.response.status === 401 ? 'unauthorized-request' : 'failure';
+
+          this.dispatchEvent(
+            new CustomEvent(event, {
```

Comment:
Firing a new event here for everything that is not a 401 also means that `nuxeo-page-provider` and `nuxeo-audit-page-provider`, which rely on `nuxeo-operation`, will be sending two events when there is an error. They already fire an `error` event when they catch the operation's error, and now they would be firing a `failure` event as well. I'd suggest we should instead fire a single `error` on these cases (see comment above), and refactor `nuxeo-page-provider` and `nuxeo-audit-page-provider`. This is also what seems to cause the `nuxeo-audit-page-provider` test to fail (see [comment](https://github.com/nuxeo/nuxeo-elements/pull/345#issue-548299867)), as there is a new even being fire that needs to be handled. To preserve the API, perhaps `nuxeo-audit-page-provider` should stop the error propagation of the error event that comes from the operation and keep firing the `notify` as it does today. WDYT @RSalem07 @nelsonsilva ?

Reply:
Thanks Gabriel for the feedbacks, I am agree with you on the solution to stop the new event `error` propagation on the two elements `nuxeo-page-provider ` and `nuxeo-audit-page-provider`, I wasn't sure if we can do this kind of stuff want I tried to fix the issue, I'll try the stuff and let you know 
thx

Reply:
Thank you for the help on cleaning this up! Please run the [pipeline](https://qa.nuxeo.org/jenkins/job/master/job/web-ui-pipeline/) once you're done, so we can be double sure nothing breaks on Web UI. 🤞